### PR TITLE
[ANCHOR-388] SEP-6: Implement RPC actions

### PR DESCRIPTION
**Pull request by philipliu**
### Description

This updates the RPC actions for SEP-6 and updates the reference server implementation to use them. This PR is quite large as it updates the unit tests for most of the RPC actions but, implementation-wise, not much has changed. It might helpful to start by reviewing the `Sep6End2EndTest`  to get a sense of what changed with this PR.

### Context

SEP-6 should support RPC API.

### Testing

- `./gradlew test`

### Known limitations

- Platform API integration tests do not test the new flow but are covered by the SEP-6 e2e tests.
- Refunds and custody service integration have not been implemented.

Both of these will be addressed by ANCHOR-508

### CODE REVIEW

**Comment by lijamie98** [comment]
Can we add class comments of the RPC handler classes? Or some reference link to the RPC definition document?

**Review by lijamie98 — APPROVED** [review]
LGTM.

**Comment by lijamie98** [comment]
At this moment, I am not familiar with the RPC enough to make too much meaningful reviews. But the PR looks fine with me.

File: core/src/main/java/org/stellar/anchor/sep6/Sep6Service.java [hunk 1]
```diff
@@ -90,6 +90,14 @@ public StartDepositResponse deposit(Sep10Jwt token, StartDepositRequest request)
             .type(request.getType())
             .assetCode(request.getAssetCode())
             .assetIssuer(asset.getIssuer())
+            // NB: these are purposely set to incorrect values.
```
[the remark this answers is not in the document]
Comment (philipliu) [reply core/src/main/java/org/stellar/anchor/sep6/Sep6Service.java:93]: It's an abbreviation for "nota bene", which basically means "note".

**Comment by philipliu** [comment]
> Can we add class comments of the RPC handler classes? Or some reference link to the RPC definition document?

Sure, I can link to the docs, but I'll do this outside of this PR.

File: core/src/main/java/org/stellar/anchor/sep6/Sep6Service.java (hunk 1 above)
Comment (Ifropc) [inline core/src/main/java/org/stellar/anchor/sep6/Sep6Service.java:93]: Does it mean there will be a state of the transaction in DB when it has incorrect amounts?

File: platform/src/main/java/org/stellar/anchor/platform/component/platform/RpcActionBeans.java [hunk 2]
```diff
@@ -165,18 +200,26 @@ NotifyOffchainFundsReceivedHandler notifyOffchainFundsReceivedHandler(
 
   @Bean
   NotifyOffchainFundsSentHandler notifyOffchainFundsSentHandler(
+      Sep6TransactionStore txn6Store,
```
Comment (Ifropc) [inline platform/src/main/java/org/stellar/anchor/platform/component/platform/RpcActionBeans.java:203]: We def should group this services together to simplify all factory methods for RPC handlers

File: platform/src/main/java/org/stellar/anchor/platform/rpc/NotifyOffchainFundsReceivedHandler.java [hunk 3]
```diff
@@ -157,9 +177,12 @@ protected void updateTransactionWithRpcRequest(
       txn.setAmountFee(request.getAmountFee().getAmount());
     }
 
-    JdbcSep24Transaction txn24 = (JdbcSep24Transaction) txn;
-    if (custodyConfig.isCustodyIntegrationEnabled()) {
-      custodyService.createTransaction(txn24);
+    // TODO: add support for SEP-6
```
Comment (Ifropc) [inline platform/src/main/java/org/stellar/anchor/platform/rpc/NotifyOffchainFundsReceivedHandler.java:180]: Is it going to be resolved in another PR?

File: core/src/main/java/org/stellar/anchor/sep6/Sep6Service.java (hunk 1 above)
Comment (philipliu) [reply core/src/main/java/org/stellar/anchor/sep6/Sep6Service.java:93]: Yeah, the amounts are technically incorrect anyway since the Anchor updates fees asynchronously. Deposit instructions and withdraw anchor accounts (to be done in the next PR) will not be provided until the business server makes a request for funds RPC so there is no way for the user to accidentally submit the wrong amount.

File: platform/src/main/java/org/stellar/anchor/platform/component/platform/RpcActionBeans.java (hunk 2 above)
Comment (philipliu) [reply platform/src/main/java/org/stellar/anchor/platform/component/platform/RpcActionBeans.java:203]: Yeah, this was quite annoying...

File: platform/src/main/java/org/stellar/anchor/platform/rpc/NotifyOffchainFundsReceivedHandler.java (hunk 3 above)
Comment (philipliu) [reply platform/src/main/java/org/stellar/anchor/platform/rpc/NotifyOffchainFundsReceivedHandler.java:180]: Yes, in the PR for ANCHOR-508.

File: core/src/main/java/org/stellar/anchor/sep6/Sep6Service.java (hunk 1 above)
Comment (Ifropc) [reply core/src/main/java/org/stellar/anchor/sep6/Sep6Service.java:93]: But if the user requests the transaction, won't we return incorrect data? Shouldn't amount just be null instead?

Comment (philipliu) [reply core/src/main/java/org/stellar/anchor/sep6/Sep6Service.java:93]: Yes. the `NotifyAmountsUpdatedHandler` validation requires the assets to be set so it can't be null ([code](https://github.com/stellar/java-stellar-anchor-sdk/blob/develop/platform/src/main/java/org/stellar/anchor/platform/rpc/NotifyAmountsUpdatedHandler.java#L53-L67)). I guess we can remove the validation?

Comment (philipliu) [reply core/src/main/java/org/stellar/anchor/sep6/Sep6Service.java:93]: I should probably think about this more, so I'll create a ticket to address this and merge it as is.

Comment (Ifropc) [reply core/src/main/java/org/stellar/anchor/sep6/Sep6Service.java:93]: I'm pretty sure RPC validation should just validate only the request having amount to not be null, if we expect transaction to start with null amounts

Comment (philipliu) [reply core/src/main/java/org/stellar/anchor/sep6/Sep6Service.java:93]: Yeah, that makes sense. Created ANCHOR-525 to fix it.
